pyvista/plotting/mapper.py

```diff
@@ -270,7 +270,7 @@ def scalar_map_mode(self) -> str:
     def scalar_map_mode(self, scalar_mode: Union[str, FieldAssociation]):
         if isinstance(scalar_mode, FieldAssociation):
             scalar_mode = scalar_mode.name
-        scalar_mode = scalar_mode.lower()
+        scalar_mode = scalar_mode.lower()  # type: ignore
```

Comment:
Why???!!! 

See https://github.com/pyvista/pyvista/pull/3318#discussion_r1067534921

Reply:
Turns out this is a Python 3.11 issue:
```
$ python3.11 -m venv .venv
$ source .venv/bin/activate
$ pip install pre-commit
$ pre-commit run --all-files mypy
```

```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

pyvista/plotting/mapper.py:273: error: Item "FieldAssociation" of "Union[str, FieldAs
sociation]" has no attribute "lower"  [union-attr]                                  
pyvista/core/composite.py:589: note: By default the bodies of untyped functions are n
ot checked, consider using --check-untyped-defs  [annotation-unchecked]             
pyvista/core/composite.py:590: note: By default the bodies of untyped functions are n
ot checked, consider using --check-untyped-defs  [annotation-unchecked]             
Found 1 error in 1 file (checked 89 source files)
```

Either we add in `type: ignore`, or I ignore the type failure on Python 3.11 until this is resolved upstream. I agree that it appears to be a red herring, but this also means that anyone else developing on Python 3.11 will encounter this weird type error as well.

Reply:
Aha! Thanks for clarifying. @adeak and I were both wondering why you were adding this. I know this is a trivial change, but could we open a dedicated PR for this to track the exact reason this is being introduced?

Reply:
> could we open a dedicated PR for this to track the exact reason this is being introduced?

Yep. Easier to track in the blame. Opening...

Reply:
Can confirm [on the mypy playground](https://mypy-play.net/?mypy=latest&python=3.10) with
```py
import enum
from typing import Union

class FieldAssociation(enum.Enum):
    POINT = 2

def foo(var: Union[str, FieldAssociation]):
    if isinstance(var, FieldAssociation):
        reveal_type(var)
        var = var.name
    reveal_type(var)
```
On mypy newest (0.991) with Python 3.10:
```
main.py:9: note: Revealed type is "__main__.FieldAssociation"
main.py:11: note: Revealed type is "builtins.str"
Success: no issues found in 1 source file
```
On Python 3.11:
```
main.py:9: note: Revealed type is "__main__.FieldAssociation"
main.py:11: note: Revealed type is "Union[builtins.str, __main__.FieldAssociation]"
Success: no issues found in 1 source file
```
This might be covered by some enum issues in https://github.com/python/mypy/issues/12841, e.g. https://github.com/python/mypy/issues/12483...

Reply:
Just need to remove it from this PR